### PR TITLE
Add regression test for GP

### PR DESCRIPTION
**Pull request by fehiepsi**
I made a regression test for SVGP based on http://gpflow.readthedocs.io/en/latest/notebooks/multiclass.html. The core of GP computations are at `conditional` and `log_prob`, so it is not necessary to test for other models. Will add other tests in the future if I find it necessary.

I also made a small fix at `matrix_triangle_solve_combat` to address "torch.trtrs has support for grad on CPU in pytorch 0.4".

I have tried to benchmark the pyro GP with gpflow and observed:
+ With the same model as in the regression test (with num_steps=20000), same dtype (float64), and the same optimizer (Adam), it took 1m for pyro GP vs 45s for gpflow on CPU. The fact that Pytorch uses inverse to solve triangular equations in multivariate distribution might be the main reason for this slowness. These might be some reasons such as: some of our computations are redundant, gpflow uses  Gaussian-Hermite quadrature to compute likelihood, gpflow uses analytic kl divergence, pyroGP uses extra_event_dims which might be slow?
+ Both pyro GP and gpflow are slower when trained on GPU (this might be expected, due to the dataset is not large). Slowness: pyro GP GPU > gpflow GPU ~ pyro GP CPU > gpflow CPU
+ The performances (loss) on both frameworks are similar.

This pull requests will be cleaned after #987 with `whiten` support.

### CODE REVIEW

File: pyro/distributions/iaf.py [hunk 1]
```diff
@@ -79,9 +79,10 @@ def _call(self, x):
         sample from the base distribution (or the output of a previous flow)
         """
         hidden = self.module.arn(x)
-        scale = self.module.sigmoid(hidden[:, 0:self.input_dim] + self.module.sigmoid_bias.type_as(hidden))
+        scale = self.module.sigmoid(hidden[:, 0:self.input_dim] +
+                                    hidden.new_tensor(self.module.sigmoid_bias))
         mean = hidden[:, self.input_dim:]
-        y = scale * x + (torch.ones(scale.shape).type_as(scale) - scale) * mean
+        y = scale * x + (scale.new_ones(scale.shape) - scale) * mean
```
[the remark this answers is not in the document]
Comment (fehiepsi) [reply pyro/distributions/iaf.py]: The reason is I have modified them without thought. ^^!

**Comment by fehiepsi** [comment]
@eb8680 I thought a regression test should be in 1 minute? The time is linear with num_steps, so which number you think is fine? We are free to change it. :)
